azurerm/resource_arm_data_factory_integration_runtime_managed.go

```diff
@@ -0,0 +1,441 @@
+			d.Set("vnet_integration", flattenArmDataFactoryIntegrationRuntimeManagedVnetIntegration(vnetProps))
+		}
+	}
+
+	if ssisProps := managedIntegrationRuntime.SsisProperties; ssisProps != nil {
+		d.Set("edition", string(ssisProps.Edition))
+		d.Set("license_type", string(ssisProps.LicenseType))
+
+		if catalogInfoProps := ssisProps.CatalogInfo; catalogInfoProps != nil {
+			d.Set("catalog_info", flattenArmDataFactoryIntegrationRuntimeManagedSsisCatalogInfo(catalogInfoProps, d))
+		}
+
+		if customSetupScriptProps := ssisProps.CustomSetupScriptProperties; customSetupScriptProps != nil {
+			d.Set("custom_setup_script", flattenArmDataFactoryIntegrationRuntimeManagedSsisCustomSetupScript(customSetupScriptProps, d))
+		}
+	}
+
+	return nil
+}
+
+func resourceArmDataFactoryIntegrationRuntimeManagedDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataFactory.IntegrationRuntimesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	factoryName := id.Path["factories"]
+	name := id.Path["integrationruntimes"]
+
+	response, err := client.Delete(ctx, resourceGroup, factoryName, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(response) {
+			return fmt.Errorf("Error deleting Data Factory Managed Integration Runtime %q (Resource Group %q, Data Factory %q): %+v", name, resourceGroup, factoryName, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmDataFactoryIntegrationRuntimeManagedComputeProperties(d *schema.ResourceData) *datafactory.IntegrationRuntimeComputeProperties {
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	computeProperties := datafactory.IntegrationRuntimeComputeProperties{
+		Location:                     &location,
+		NodeSize:                     utils.String(d.Get("node_size").(string)),
+		NumberOfNodes:                utils.Int32(int32(d.Get("number_of_nodes").(int))),
+		MaxParallelExecutionsPerNode: utils.Int32(int32(d.Get("max_parallel_executions_per_node").(int))),
+	}
+
+	if _, ok := d.GetOk("vnet_integration"); ok {
+		vnetProps := d.Get("vnet_integration").([]interface{})[0].(map[string]interface{})
+		computeProperties.VNetProperties = &datafactory.IntegrationRuntimeVNetProperties{
+			VNetID: utils.String(vnetProps["vnet_id"].(string)),
+			Subnet: utils.String(vnetProps["subnet_name"].(string)),
+		}
+	}
+
+	return &computeProperties
+}
+
+func expandArmDataFactoryIntegrationRuntimeManagedSsisProperties(d *schema.ResourceData) *datafactory.IntegrationRuntimeSsisProperties {
+	ssisProperties := &datafactory.IntegrationRuntimeSsisProperties{
+		Edition:     datafactory.IntegrationRuntimeEdition(d.Get("edition").(string)),
+		LicenseType: datafactory.IntegrationRuntimeLicenseType(d.Get("license_type").(string)),
+	}
+
+	if _, ok := d.GetOk("catalog_info"); ok {
+		catalogInfo := d.Get("catalog_info").([]interface{})[0].(map[string]interface{})
```

Comment:
And here we should confirm that `d.Get("catalog_info").([]interface{})[0]` isn't nil

Reply:
Fixed